pkg/xds/experimental/client/listener.go

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package client
+
+import (
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/envoy/xds"
+)
+
+// listener is a helper structure for watchers focused on handling a single, registered callback.
+type listener struct {
+	ID      uint64
+	typeUrl string
+	cb      WatcherCallback
+	trigger chan struct{}
+
+	log       *slog.Logger
+	resources xds.ResourceSource
+}
+
+var _ xds.ResourceVersionObserver = (*listener)(nil)
+
+// HandleNewResourceVersion implements xds.ResourceVersionObserver.
+// It triggers the callback process.
+func (l *listener) HandleNewResourceVersion(typeUrl string, _ uint64) {
+	if typeUrl != l.typeUrl {
+		return
```

Comment:
this would be highly unusual -- an error should be logged.

Reply:
Done.